status-im/status-react
# Desktop notification on new message received

- Desktop notifications turn on/off tumbler is added to user Profile page
- user profile is extended with desktop-notifications? boolean property
- Checks if desktop notification emitting required on every new message is adding to chat
- react-native-desktop-notification native module based on snorenotify
----
- Application is renamed from StatusIm to Status.

Renaming of app is related to issues with emitting desktop notifications on MacOS. Recently, the app was renamed there from StatusIm to Status. Current changes renames the native app completely.

Depends on changes in https://github.com/status-im/react-native-desktop/pull/356

BREAKING CHANGE: following the current internal logic, app will start reading/writing db and logs data to folder called `Status` instead of `StatusIm`. Manual action to rename available folder should be taken in advance.

----

PR is based on @siphiuel 's work https://github.com/status-im/status-react/pull/5330

and on initial https://github.com/status-im/status-react/pull/5129

### Summary:

Basic support of desktop notifications

### Review notes:

- react-native-desktop-notification native module introduced
- Fork of SnoreNotify library ( https://github.com/status-im/snorenotify.git ) used as backend to cover all notifications related functionality
- appimagetool is used to produce AppImage on Jenkins

### Testing notes (optional):
- Current PR introduces receiving of notifications for any incoming message even if app window is opened

### Steps to test:
- Open Status on primary testing machine
- Navigate to some chat
- Open another Status instance on another device
- Send the message into chat opened on primary testing machine
- Desktop notification "New Message" with message content should be triggered

status: ready

Usage of snorenotify introduced dependency on additional dev package to be installed in development environment:

Linux:
sudo apt install extra-cmake-modules

MacOS:
brew install kde-mac/kde/kf5-extra-cmake-modules

----

Implementation summary:

- MacOS bundle security check can flag usage of linkage of executable to 3rd party .dylib without any obvious reason. Signing tool successful validation doesn't catch the reason of security issue.
- Because of above issues, SnoreNotify libs and plugins were linked statically.
